.github/actions/python-environment/action.yml

```diff
@@ -32,24 +32,45 @@ runs:
   using: "composite"
   steps:
 
-    - name: Set up pipx if not present
+
+    - name: Set up Python (${{ inputs.python-version}})
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    - name: Ensure curl
+      shell: bash
+      run: sudo apt update && sudo apt install -y wget
+
+    - name: Set up Poetry (${{ inputs.poetry-version }})
       shell: bash
       run: |
-        sudo apt update && sudo apt install -y pipx
-        python3 -m pipx ensurepath
+        wget --tries=10 --waitretry 10 --retry-connrefused --https-only -O - https://install.python-poetry.org | POETRY_VERSION="${{ inputs.poetry-version }}" "$PYTHON_BINARY" -
         echo "$HOME/.local/bin" >> $GITHUB_PATH
+      env:
+        PYTHON_BINARY: "python${{ inputs.python-version }}"
 
-    - name: Set up Poetry (${{ inputs.poetry-version }})
+    - name: Setup cache variables
+      id: setup-cache-variables
       shell: bash
-      run: pipx install poetry==${{ inputs.poetry-version }}
+      run: |
+        echo "ImageOS=$ImageOS"
+        echo "ImageVersion=$ImageVersion"
+        POETRY_ENV_PATH=$(poetry config virtualenvs.path)
+        echo "POETRY_ENV_PATH=$POETRY_ENV_PATH" >>  $GITHUB_OUTPUT
+        echo "IMAGE_OS=$ImageOS" >> $GITHUB_OUTPUT
+        echo "IMAGE_VERSION=$ImageVersion" >> $GITHUB_OUTPUT
 
-    - name: Set up Python (${{ inputs.python-version}})
-      uses: actions/setup-python@v5
+    - name: Cache Poetry environment
+      if: inputs.use-cache == 'true'
+      id: cache-poetry-env
+      uses: actions/cache@v4
       with:
-        python-version: ${{ inputs.python-version }}
-        cache: ${{ inputs.use-cache == 'true' && 'poetry' || '' }}
+        path: ${{ steps.setup-cache-variables.outputs.POETRY_ENV_PATH }}
+        key: poetry-env-${{ steps.setup-cache-variables.outputs.IMAGE_OS }}-${{ steps.setup-cache-variables.outputs.IMAGE_VERSION }}-${{ inputs.poetry-version }}-${{ inputs.python-version }}-${{ inputs.extras }}
 
     - name: Poetry install with extras
+      if: inputs.use-cache != 'true' || steps.cache-poetry-env.outputs.cache-hit != 'true'
       working-directory: ${{ inputs.working-directory }}
       shell: bash
       run: |
```

Comment:
I believe the EXTRAS environment variables is not set, check https://github.com/exasol/python-toolbox/commit/b6972497ea84b1c8bd3ab1de46742e6782365cbb
@ArBridgeman

Reply:
Actually, we will need the extra installs quite soon also for notebook-connector

Reply:
I added the old line then back